qiskit/transpiler/__init__.py

```diff
@@ -40,714 +41,1132 @@
-      this gate must be decomposed.  This decomposition is quite costly:
+   .. plot::
+      :include-source:
 
-      .. plot::
-         :include-source:
+      from qiskit.circuit import QuantumCircuit
 
-         from qiskit import QuantumCircuit
+      ccx_circ = QuantumCircuit(3)
+      ccx_circ.ccx(0, 1, 2)
+      ccx_circ.decompose().draw(output='mpl')
 
-         ccx_circ = QuantumCircuit(3)
-         ccx_circ.ccx(0, 1, 2)
-         ccx_circ.decompose().draw(output='mpl')
+   For every Toffoli gate in a quantum circuit, the hardware may execute up to six CNOT
+   gates, and a handful of single-qubit gates.  From this example, it should be
+   clear that any algorithm that makes use of multiple Toffoli gates will end up as a
+   circuit with large depth and will therefore be appreciably affected by noise and gate
+   errors.
 
-      For every Toffoli gate in a quantum circuit, the IBM Quantum hardware may execute up to
-      six CNOT gates, and a handful of single-qubit gates.  From this example, it should be
-      clear that any algorithm that makes use of multiple Toffoli gates will end up as a
-      circuit with large depth and will therefore be appreciably affected by noise and gate
-      errors.
 
+.. _layout_stage:
 
-   .. raw:: html
+Layout Stage
+------------
 
-      <br>
+Quantum circuits are abstract entities whose qubits are "virtual" representations of actual
+qubits used in computations.  We need to be able to map these virtual qubits in a one-to-one
+manner to the "physical" qubits in an actual quantum device.
 
-.. dropdown:: Initial layout
-   :animate: fade-in-slide-down
+.. image:: /source_images/mapping.png
 
-   Quantum circuits are abstract entities whose qubits are "virtual" representations of actual
-   qubits used in computations.  We need to be able to map these virtual qubits in a one-to-one
-   manner to the "physical" qubits in an actual quantum device.
+By default, qiskit will do this mapping for you.  The choice of mapping depends on the
+properties of the circuit, the particular device you are targeting, and the optimization
+level that is chosen The choice of initial layout is extremely important when:
 
-   .. image:: /source_images/mapping.png
+1. Computing the number of swap operations needed to map the input circuit onto the device
+   topology.
 
-   .. raw:: html
+2. Taking into account the noise properties of the device.
 
-      <br><br>
+The choice of `initial_layout` can mean the difference between getting a result,
+and getting nothing but noise. Due to the importance of this stage the preset pass managers
+try a few different methods to find the best layout. Typically this involves 2 stages, first
+trying to find a "perfect" layout (a layout which does not require any swap operations) and
+a heuristic pass that tries to find the best layout to use if a perfect layout can not be found.
+For the first stage there are 2 passes typically used for this:
```

Comment:
Is there another word we can use to describe these? They're not strictly separate "stages", correct? Maybe phases, or strategies?

Reply:
Yeah "stages" is a poor choice here because they're not "stages" as in the preset pass manager. "Phases" might be a good word, I'm thinking maybe "steps" would work too.